webui/react/src/utils/flatRun.ts

```diff
@@ -0,0 +1,71 @@
+import { deletableRunStates, killableRunStates, terminalRunStates } from 'constants/states';
+import { PermissionsHook } from 'hooks/usePermissions';
+import { FlatRun, FlatRunAction } from 'types';
+
+type FlatRunChecker = (flatRun: Readonly<FlatRun>) => boolean;
+
+type FlatRunPermissionSet = Pick<
+  PermissionsHook,
+  'canCreateExperiment' | 'canDeleteFlatRun' | 'canModifyExperiment' | 'canMoveFlatRun'
+>;
+
+const flatRunCheckers: Record<FlatRunAction, FlatRunChecker> = {
+  [FlatRunAction.Archive]: (flatRun) =>
+    !flatRun.parentArchived && !flatRun.archived && terminalRunStates.has(flatRun.state),
+
+  [FlatRunAction.Delete]: (flatRun) => deletableRunStates.has(flatRun.state),
+
+  [FlatRunAction.Kill]: (flatRun) => killableRunStates.includes(flatRun.state),
+
+  [FlatRunAction.Move]: (flatRun) => !flatRun.parentArchived && !flatRun.archived,
+
+  // [FlatRunAction.Pause]: (run) => pausableRunStates.has(run.state),
+
+  [FlatRunAction.Unarchive]: (flatRun) => terminalRunStates.has(flatRun.state) && flatRun.archived,
+};
+
+export const canActionFlatRun = (action: FlatRunAction, flatRun: Readonly<FlatRun>): boolean => {
+  return flatRunCheckers[action](flatRun);
+};
+
+const getActionsForFlatRun = (
```

Comment:
nit: already changing this in my PR but, `export`

Reply:
fixed